File: src/ExactOnline.Client.Models/Attributes/DataServiceKey.vb
' Simulates the actual class used in the API
Public Class DataServiceKey
	Inherits Attribute

	Private _dataServiceKey As String


	Public Sub New(dataServiceKey As String)
		_dataServiceKey = dataServiceKey
	End Sub

	Public Sub New(dataServiceKey As String, dataServiceKey2 As String)
		_dataServiceKey = dataServiceKey
	End Sub

	Public Sub New(dataServiceKey As String, dataServiceKey2 As String, dataServiceKey3 As String)
		_dataServiceKey = dataServiceKey
	End Sub

	Public Property DataServiceKey() As String
		Get
			Return _dataServiceKey
		End Get
		Set(Value As String)
			_dataServiceKey = Value
		End Set
	End Property
End Class

File: src/ExactOnline.Client.Models/Attributes/DataServiceKey.cs
using System;

using Newtonsoft.Json;

public class DataServiceKey : Attribute
{
    public DataServiceKey(string dataServiceKey)
    {
        DataServiceKeyName = dataServiceKey;
    }

    public DataServiceKey(string dataServiceKey, string dataServiceKey2)
    {
        DataServiceKeyName = dataServiceKey;
    }

    public DataServiceKey(string dataServiceKey, string dataServiceKey2, string dataServiceKey3)
    {
        DataServiceKeyName = dataServiceKey;
    }

    [JsonProperty(PropertyName = "dataServiceKey")]
    public string DataServiceKeyName { get; set; }
}
